SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -201,48 +229,77 @@ void USpatialNetDriver::InitiateConnectionToSpatialOS(const FURL& URL)
 
 	if (!bPersistSpatialConnection)
 	{
-		// Destroy the old connection
 		GameInstance->DestroySpatialWorkerConnection();
-
-		// Create a new SpatialWorkerConnection in the SpatialGameInstance.
 		GameInstance->CreateNewSpatialWorkerConnection();
 	}
 
 	Connection = GameInstance->GetSpatialWorkerConnection();
 
-	if (URL.HasOption(TEXT("locator")))
+	bool bShouldLoadFromURL = true;
+	bool bUseReceptionist = true;
+
+	// If this is the first connection try using the command line arguments to setup the config objects.
+	// If arguments can not be found we will use the regular flow of loading from the input URL.
+	if (!GameInstance->bFirstConnectionToSpatialOSAttempted || URL.Host == SpatialConstants::RECONNECT_USING_ARGUMENTS)
 	{
-		// Obtain PIT and LT.
-		Connection->LocatorConfig.PlayerIdentityToken = URL.GetOption(TEXT("playeridentity="), TEXT(""));
-		Connection->LocatorConfig.LoginToken = URL.GetOption(TEXT("login="), TEXT(""));
-		Connection->LocatorConfig.UseExternalIp = true;
-		Connection->LocatorConfig.WorkerType = GameInstance->GetSpatialWorkerType().ToString();
+		FString CommandLineLocatorHost;
+		FParse::Value(FCommandLine::Get(), TEXT("locatorHost"), CommandLineLocatorHost);
+		if(!CommandLineLocatorHost.IsEmpty())
+		{
+			bShouldLoadFromURL = !Connection->LocatorConfig.TryLoadCommandLineArgs();
+			bUseReceptionist = false;
+		}
+		else
+		{
+			bShouldLoadFromURL = !Connection->ReceptionistConfig.TryLoadCommandLineArgs();
+			bUseReceptionist = true;
+		}
+
+		GameInstance->bFirstConnectionToSpatialOSAttempted = true;
 	}
-	else // Using Receptionist
-	{
-		Connection->ReceptionistConfig.WorkerType = GameInstance->GetSpatialWorkerType().ToString();
 
-		// Check for overrides in the travel URL.
-		if (!URL.Host.IsEmpty() && URL.Host.Compare(SpatialConstants::LOCAL_HOST) != 0)
+	// Copy relevant URL info into the relevant config object.
+	if(bShouldLoadFromURL)
+	{
+		if (URL.Host == SpatialConstants::LOCATOR_HOST || URL.HasOption(TEXT("locator")))
 		{
-			Connection->ReceptionistConfig.ReceptionistHost = URL.Host;
+			FLocatorConfig& LocatorConfig = Connection->LocatorConfig;
+			LocatorConfig.PlayerIdentityToken = URL.GetOption(*SpatialConstants::URL_PLAYER_IDENTITY_OPTION, TEXT(""));
+			LocatorConfig.LoginToken = URL.GetOption(*SpatialConstants::URL_LOGIN_OPTION, TEXT(""));
+			bUseReceptionist = false;
 		}
+		else
+		{
+			Connection->ReceptionistConfig.SetReceptionistHost(URL.Host);
+			bUseReceptionist = true;
+		}
+	}
+
+	// Setup for the config objects regardless of loading from command line or URL
+	if (bUseReceptionist)
```

Comment:
Maybe we should break this whole InitiateConnectionToSpatialOS into seperate functions. 
Using `InitiateConnectionToSpatialOS(const FURL& URL)` with a URL seems strange when it will just use command line args.
Perhaps a `InitiateConnectionToSpatialOS(const FURL& URL)` and ``InitiateConnectionToSpatialOSCommandLineArgs()` (rouge idea)
And then break up the `if (bUseReceptionist)` into `ConnectWithReceptionist(ReceptionistConfig Config)` and `ConnectWithLocator(LocatorConfig Config)`.
If that make sense.

Reply:
Kinda done. Not super happy with it but you can take a look.